Objects/tupleobject.c

```diff
@@ -76,24 +79,15 @@ _PyTuple_DebugMallocStats(FILE *out)
 #endif
 }
 
-PyObject *
-PyTuple_New(Py_ssize_t size)
+static PyObject *
+tuple_new_internal(Py_ssize_t size)
```

Comment:
I'm not comfortable with all these functions called "tuple_new". Usually, "new" returns an initialized/ready object. Here it's a half baked object. I propose to rename the function to "tuple_alloc".

Since creating a tuple now requires multiple function calls, I would prefer to see a short comment to explain the purpose of each function: explain what we get.

Here I understand that it's a helper function, it must not be called directly, the tuple is not tracked by the GC.